src/Backups/BackupCoordinationStageSync.cpp

```diff
@@ -17,42 +18,59 @@ namespace ErrorCodes
 }
 
 
-BackupCoordinationStageSync::BackupCoordinationStageSync(const String & zookeeper_path_, zkutil::GetZooKeeper get_zookeeper_, Poco::Logger * log_)
-    : zookeeper_path(zookeeper_path_)
+BackupCoordinationStageSync::BackupCoordinationStageSync(CoordinationSettings settings_, zkutil::GetZooKeeper get_zookeeper_, Poco::Logger * log_)
+    : zookeeper_path(settings_.root_zookeeper_path + "/stage")
     , get_zookeeper(get_zookeeper_)
     , log(log_)
 {
+    zookeeper_retries_info = ZooKeeperRetriesInfo(
+        "BackupCoordinationStageSync",
+        log,
+        settings_.max_retries,
+        settings_.initial_backoff_ms,
+        settings_.max_backoff_ms);
     createRootNodes();
 }
 
 void BackupCoordinationStageSync::createRootNodes()
 {
-    auto zookeeper = get_zookeeper();
-    zookeeper->createAncestors(zookeeper_path);
-    zookeeper->createIfNotExists(zookeeper_path, "");
+    ZooKeeperRetriesControl retries_ctl("createRootNodes", zookeeper_retries_info);
+    retries_ctl.retryLoop([&]()
+    {
+        auto zookeeper = get_zookeeper();
+        zookeeper->createAncestors(zookeeper_path);
+        zookeeper->createIfNotExists(zookeeper_path, "");
+    });
 }
 
 void BackupCoordinationStageSync::set(const String & current_host, const String & new_stage, const String & message)
 {
-    auto zookeeper = get_zookeeper();
-
-    /// Make an ephemeral node so the initiator can track if the current host is still working.
-    String alive_node_path = zookeeper_path + "/alive|" + current_host;
-    auto code = zookeeper->tryCreate(alive_node_path, "", zkutil::CreateMode::Ephemeral);
-    if (code != Coordination::Error::ZOK && code != Coordination::Error::ZNODEEXISTS)
-        throw zkutil::KeeperException(code, alive_node_path);
-
-    zookeeper->createIfNotExists(zookeeper_path + "/started|" + current_host, "");
-    zookeeper->create(zookeeper_path + "/current|" + current_host + "|" + new_stage, message, zkutil::CreateMode::Persistent);
+    ZooKeeperRetriesControl retries_ctl("set", zookeeper_retries_info);
+    retries_ctl.retryLoop([&]()
+    {
+        auto zookeeper = get_zookeeper();
+
+        /// Make an ephemeral node so the initiator can track if the current host is still working.
+        String alive_node_path = zookeeper_path + "/alive|" + current_host;
+        /// FIXME: alive_node_path should be ephemeral to determine whether the replica is dead or not
+        /// But since we retry all the operations we cannot rely on it.
+        zookeeper->createIfNotExists(alive_node_path, "");
```

Comment:
Probably the test `test_backup_restore_on_cluster/test.py::test_stop_other_host_during_backup` is going to fail, we can disable it